wgpu-core/src/device/mod.rs

```diff
@@ -2575,3 +2618,30 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         buffer.map_state = resource::BufferMapState::Idle;
     }
 }
+
+fn validate_shader(module: &naga::Module, entry_point_name: &str, execution_model: ExecutionModel) {
```

Comment:
It would be great to start writing this in a way that propagates errors. After all, we don't want to have logs and assertions for the medium/far future.
```rust
enum ShaderValidationError {
  MissingEntryPoint,
  WrongExecutionModel(ExecutionModel),
}

fn validate_shader(...) -> Result<(), ShaderValidationError> {
}

...
            if let Some(ref module) = shader_module.module {
                if let Err(e) = validate_shader(module, entry_point_name, ExecutionModel::GLCompute) {
                  log::error!("Failed validating {}: {:?}", entry_point_name, e);
                }
            }
```

Then we'll start returning errors from `create_render_pipeline` and such, and we'll be able to simply forward your new error.